vscode/src/local-context/local-embeddings.ts

```diff
@@ -7,45 +7,56 @@ import {
     type ContextGroup,
     type ContextStatusProvider,
     type EmbeddingsSearchResult,
+    FeatureFlag,
     type FileURI,
     type LocalEmbeddingsFetcher,
     type LocalEmbeddingsProvider,
+    featureFlagProvider,
     isDotCom,
     isFileURI,
     uriBasename,
 } from '@sourcegraph/cody-shared'
 
+import type { EmbeddingsProvider } from '@sourcegraph/cody-shared/src/codebase-context/context-status'
 import type { IndexHealthResultFound, IndexRequest } from '../jsonrpc/embeddings-protocol'
 import type { MessageHandler } from '../jsonrpc/jsonrpc'
 import { logDebug } from '../log'
 import { captureException } from '../services/sentry/sentry'
 import { CodyEngineService } from './cody-engine'
 
-export function createLocalEmbeddingsController(
+export async function createLocalEmbeddingsController(
     context: vscode.ExtensionContext,
     config: LocalEmbeddingsConfig
-): LocalEmbeddingsController {
-    return new LocalEmbeddingsController(context, config)
+): Promise<LocalEmbeddingsController> {
+    const useSourcegraphEmbeddings = await featureFlagProvider.evaluateFeatureFlag(
+        FeatureFlag.CodyUseSourcegraphEmbeddings
+    )
+    return new LocalEmbeddingsController(
+        context,
+        config,
+        useSourcegraphEmbeddings ? sourcegraphModelConfig : openaiModelConfig
+    )
 }
 
-export type LocalEmbeddingsConfig = Pick<
-    ConfigurationWithAccessToken,
-    'serverEndpoint' | 'accessToken'
-> & {
-    testingLocalEmbeddingsModel: string | undefined
-    testingLocalEmbeddingsDimension: number | undefined
-    testingLocalEmbeddingsEndpoint: string | undefined
-    testingLocalEmbeddingsIndexLibraryPath: string | undefined
-}
+export type LocalEmbeddingsConfig = Pick<ConfigurationWithAccessToken, 'serverEndpoint' | 'accessToken'>
 
-function getIndexLibraryPath(): FileURI {
+function getIndexLibraryPath(modelSuffix: string): FileURI {
     switch (process.platform) {
         case 'darwin':
-            return URI.file(`${process.env.HOME}/Library/Caches/com.sourcegraph.cody/embeddings`)
+            return URI.file(
+                `${process.env.HOME}/Library/Caches/com.sourcegraph.cody/embeddings` +
+                    (modelSuffix !== '' ? '/' + modelSuffix : '')
```

Comment:
You could simplify this a bit by reversing the branches.

```suggestion
                    (modelSuffix == '' ? '' : '/' + modelSuffix)
```